pkg/mesh/mesh.go

```diff
@@ -774,19 +795,31 @@ func linkByIndex(index int) (netlink.Link, error) {
 
 // updateNATEndpoints ensures that nodes and peers behind NAT update
 // their endpoints from the WireGuard configuration so they can roam.
-func updateNATEndpoints(nodes map[string]*Node, peers map[string]*Peer, conf *wireguard.Conf) {
+func updateNATEndpoints(nodes map[string]*Node, peers map[string]*Peer, conf *wireguard.Conf, logger log.Logger) map[string]*wireguard.Endpoint {
```

Comment:
Let's also update the name and comment of this function to reflect the new behavior, e.g.:
```
discoverNATEndpoints uses the node's WireGuard configuration to returns a list of the most recently discovered endpoints for all nodes and peers behind NAT so that they can roam.
```

Reply:
done